javascript/i18n/phonenumbers/demo.js

```diff
@@ -46,75 +46,82 @@ function phoneNumberParser() {
     output.append('\nResult from isPossibleNumber(): ');
     output.append(isPossible);
-    if (!isPossible) {
-      output.append('\nResult from isPossibleNumberWithReason(): ');
-      var PNV = i18n.phonenumbers.PhoneNumberUtil.ValidationResult;
-      switch (phoneUtil.isPossibleNumberWithReason(number)) {
-        case PNV.INVALID_COUNTRY_CODE:
-          output.append('INVALID_COUNTRY_CODE');
-          break;
-        case PNV.TOO_SHORT:
-          output.append('TOO_SHORT');
-          break;
-        case PNV.TOO_LONG:
-          output.append('TOO_LONG');
-          break;
-      }
-      // IS_POSSIBLE shouldn't happen, since we only call this if _not_
-      // possible.
-      output.append('\nNote: numbers that are not possible have type ' +
-          'UNKNOWN, an unknown region, and are considered invalid.');
-    } else {
-      var isNumberValid = phoneUtil.isValidNumber(number);
-      output.append('\nResult from isValidNumber(): ');
-      output.append(isNumberValid);
-      if (isNumberValid && regionCode && regionCode != 'ZZ') {
-        output.append('\nResult from isValidNumberForRegion(): ');
-        output.append(phoneUtil.isValidNumberForRegion(number, regionCode));
-      }
-      output.append('\nPhone Number region: ');
-      output.append(phoneUtil.getRegionCodeForNumber(number));
-      output.append('\nResult from getNumberType(): ');
-      var PNT = i18n.phonenumbers.PhoneNumberType;
-      switch (phoneUtil.getNumberType(number)) {
-        case PNT.FIXED_LINE:
-          output.append('FIXED_LINE');
-          break;
-        case PNT.MOBILE:
-          output.append('MOBILE');
-          break;
-        case PNT.FIXED_LINE_OR_MOBILE:
-          output.append('FIXED_LINE_OR_MOBILE');
-          break;
-        case PNT.TOLL_FREE:
-          output.append('TOLL_FREE');
-          break;
-        case PNT.PREMIUM_RATE:
-          output.append('PREMIUM_RATE');
-          break;
-        case PNT.SHARED_COST:
-          output.append('SHARED_COST');
-          break;
-        case PNT.VOIP:
-          output.append('VOIP');
-          break;
-        case PNT.PERSONAL_NUMBER:
-          output.append('PERSONAL_NUMBER');
-          break;
-        case PNT.PAGER:
-          output.append('PAGER');
-          break;
-        case PNT.UAN:
-          output.append('UAN');
-          break;
-        case PNT.UNKNOWN:
-          output.append('UNKNOWN');
-          break;
-      }
+    output.append('\nResult from isPossibleNumberWithReason(): ');
+    var PNV = i18n.phonenumbers.PhoneNumberUtil.ValidationResult;
+    switch (phoneUtil.isPossibleNumberWithReason(number)) {
+      case PNV.INVALID_COUNTRY_CODE:
+        output.append('INVALID_COUNTRY_CODE');
+        break;
+      case PNV.TOO_SHORT:
+        output.append('TOO_SHORT');
+        break;
+      case PNV.TOO_LONG:
+        output.append('TOO_LONG');
+        break;
+      case PNV.IS_POSSIBLE:
```

Comment:
I would break this up:
if possible, then say either nothing, or "N.B. LOCAL_ONLY" if that applies. If not possible, keep it as before but add the extra case statement for INVALID_LENGTH.

Reply:
Got it. It makes sense. As isValid() return false for local only number, should I stop displaying those? Like this:
https://screenshot.googleplex.com/BRXu6Lj3t2uuHwY
Not sure, I prefer displaying isValid() results also to provide more info as users may not have an idea about the hierarchy.